compiler/rustc_mir/src/transform/validate.rs

```diff
@@ -294,6 +294,26 @@ impl<'a, 'tcx> Visitor<'tcx> for TypeChecker<'a, 'tcx> {
                     );
                 }
             }
+            StatementKind::CopyNonOverlapping(box rustc_middle::mir::CopyNonOverlapping {
+                ref src,
+                ref dst,
+                ref count,
+            }) => {
+                let op_src_ty =
+                    src.ty(&self.body.local_decls, self.tcx).builtin_deref(true).unwrap().ty;
+                let op_dst_ty =
+                    dst.ty(&self.body.local_decls, self.tcx).builtin_deref(true).unwrap().ty;
```

Comment:
Since we are validating MIR for correctness and reporting (somewhat) nice errors, I think these unwraps should also use the `self.fail` mechanism instead.